core/camel-support/src/main/java/org/apache/camel/support/cache/DefaultProducerCache.java

```diff
@@ -173,7 +175,7 @@ public void releaseProducer(Endpoint endpoint, AsyncProducer producer) {
     @Override
     public Exchange send(Endpoint endpoint, Exchange exchange, Processor resultProcessor) {
         if (camelContext.isStopped()) {
-            exchange.setException(new RejectedExecutionException("CamelContext is stopped"));
+            exchange.setException(REJECTED_EXECUTION_EXCEPTION);
```

Comment:
Warning if you do so, the stack trace won't be correct anymore so it will make the debugging more complex

Reply:
That's a good point. I wonder if we could get away with it by using a better exception message. 

@davsclaus any thoughts about this?

Reply:
We could probably instantiate it here, instead of in the declaration. That way it would retain the stack trace.

Reply:
Or just call `fillInStackTrace` once.

Reply:
I changed it to fill the stack trace once, so I guess this should do the trick.

Reply:
Interresting optimization - however its ONLY when camel is stopped so I wonder what you see in those test that makes it worth to do this? in other words its only if the user tries to use the producer after camel has been stopped.

Reply:
This doesn't affect the hot part of the test, but I did notice that it was creating a lot of objects like this during the stop (under load, using main with `camel.main.durationMaxMessages`). 

So I guess this tends to occur when we have a lot of exchanges in transit and something has triggered the shutdown?

Reply:
Obs.: this one is pretty minor and I think we can get rid of it if you guys think this is not worth the effort. I don't think this one matters much in the big picture.